docs/reference/mapping/params/eager-global-ordinals.asciidoc

```diff
@@ -0,0 +1,76 @@
+[[eager-global-ordinals]]
+=== `eager_global_ordinals`
+
+Global ordinals is a data-structure on top of doc values, that maintains an
+incremental numbering for each unique term in a lexicographic order. Each
+term has a unique number and the number of term 'A' is lower than the
+number of term 'B'. Global ordinals are only supported on
+<<keyword,`keyword`>> fields, as well as <<text,`text`>> fields when
+<<fielddata,`fielddata`>> is enabled.
+
+Doc values (and fielddata) also have ordinals, which is a unique numbering for
+all terms in a particular segment and field. Global ordinals just build on top
+of this, by providing a mapping between the segment ordinals and the global
+ordinals, the latter being unique across the entire shard.
+
+Global ordinals are used for features that use segment ordinals, such as
+the <<search-aggregations-bucket-terms-aggregation,`terms` aggregation>>,
+to improve the execution time. A terms aggregation relies purely on global
+ordinals to perform the aggregation at the shard level, then converts global
+ordinals to the real term only for the final reduce phase, which combines
+results from different shards.
+
+Global ordinals for a specified field are tied to _all the segments of a
+shard_, while doc values ordinals are tied to a single segment. This is
+different than for field data for a specific field which is tied to a single
+segment. For this reason global ordinals need to be entirely rebuilt
+whenever a once new segment becomes visible.
+
+The loading time of global ordinals depends on the number of terms in a field,
+but in general it is low, since it source field data has already been loaded.
+The memory overhead of global ordinals is a small because it is very
+efficiently compressed.
+
+By default, global ordinals are loaded at search-time, which is the right
+trade-off if you are optimizing for indexing speed. However, if you are more
+interested in search speed, it could be interesting to set
+`eager_global_ordinals: true` on fields that you plan to use in terms
+aggregations:
+
+[source,js]
+------------
+PUT my_index/_mapping/my_type
+{
+  "properties": {
+    "tags": {
+      "type": "keyword",
+      "eager_global_ordinals": true
+    }
+  }
+}
+------------
+// CONSOLE
+// TEST[s/^/PUT my_index\n/]
+
+This will shift the cost from search-time to refresh-time, ie. Elasticsearch
+will make sure that global ordinals are built before publishing updates to the
+content of the index.
+
+If you ever decide that you do not need to run `terms` aggregations on this
+field anymore, then you can disable eager loading of global ordinals:
```

Comment:
I think this is missing an "at any time" somewhere. Folks are used to mapping things being fixed at field creation but this isn't.